tool_path_planner/include/tool_path_planner/surface_walk_raster_generator.h

```diff
@@ -163,7 +162,6 @@ class SurfaceWalkRasterGenerator : public PathGenerator
   };
 
   SurfaceWalkRasterGenerator() = default;
-  virtual ~SurfaceWalkRasterGenerator() = default;
```

Comment:
Could you explain the reasoning behind removing this line?

Reply:
Don't need to define it if it doesn't do anything. Also it hides rather than overrides the base class destructor since the `override` keyword isn't present

Reply:
If I understand correctly, you never really override a destructor: the destructor of an inherited class calls the destructor of its base class after exiting.  Making the destructor virtual just ensures that if something ever inherits from class A, it will correctly call class A's destructor in turn.  (Not the most authoritative source, but [here](https://www.geeksforgeeks.org/virtual-destructor/) you go.)

Reply:
As long as the base class has it you should be good. but

> From discussions with @drchrislewis @jrgnicho @DavidMerzJr, the output structure of the tool paths are neither documented nor entirely intuitive. I propose renaming them in a way more understandable way and with notes about what each level represents:
> 
>     * `ToolPathSegment` -> `RasterSegment`
>       
>       * **Contiguous series of waypoints generated from a single "slice" through a mesh**
> 
>     * `ToolPath` -> `Raster`
>       
>       * **Set of raster segments that lie on the the same "slice" through a mesh, but are separated by some distance (like on either side of a hole)**
>       * I didn't initially think to represent a tool path like this, but I understand the value and don't propose eliminating this layer of the tool path
>       * IMO, the original name (`ToolPath`) is a misnomer because it implies that it represents a set of rasters across the entire surface
> 
>     * `ToolPaths` -> `ToolPath`
>       
>       * **Set of rasters generated from different "slices" through a mesh**
>       * IMO, the original name (`ToolPaths`) is a misnomer because it implies that it represents multiple sets of raster patterns across full surface. It also implies that multiple tool paths will be returned, even though only one mesh can be used as input

My preference would be to keep toolpath. It is a general definition which can represent a broader set specialize toolpaths like (Raster, ZigZag, Contour/Edge, etc.). This is also the terminology used by CNC. The definition of toolpath, is the path through space which a tool follows. The term raster, is a specific representation of a toolpath which is defined as a rectangular pattern of parallel lines. This will allow the same data structure to be used more broadly through since it not only includes Raster,ZigZag, and Edge generation.